bdk_electrum_example/src/electrum.rs

```diff
@@ -0,0 +1,205 @@
+        // Add them into the KeyChainScan
+        for (keychain, mut scripts) in scripts {
+            let mut last_active_index = 0;
+            let mut unused_script_count = 0usize;
+
+            while let Some((index, script)) = scripts.next() {
+                let history = self
+                    .script_get_history(&script)?
+                    .iter()
+                    .map(|history_result| {
+                        if history_result.height > 0 && (history_result.height as u32) <= tip_height
+                        {
+                            return (
+                                history_result.tx_hash,
+                                TxHeight::Confirmed(history_result.height as u32),
+                            );
+                        } else {
+                            return (history_result.tx_hash, TxHeight::Unconfirmed);
+                        };
+                    })
+                    .collect::<Vec<(Txid, TxHeight)>>();
+
+                if history.is_empty() {
+                    unused_script_count += 1;
+                } else {
+                    last_active_index = index;
+                    unused_script_count = 0;
+
+                    for (txid, index) in history {
+                        if let Err(err) = sparse_chain.insert_tx(txid, index) {
+                            match err {
+                                InsertTxErr::TxTooHigh => {
+                                    /* We should not encounter this error as we ensured TxHeight <= tip_height */
+                                    unreachable!();
+                                }
+                                InsertTxErr::TxMoved => {
+                                    /* This means there is a reorg, we will handle this situation below */
+                                }
+                            }
+                        }
+                    }
+                }
+
+                if unused_script_count >= stop_gap.unwrap_or(usize::MAX) {
+                    break;
+                }
+            }
+
+            keychain_index_update.insert(keychain, last_active_index);
+        }
+
+        // To detect reorgs during syncing we re-apply last known
+        // 20 blocks again into sparsechain
+        // TODO: Handle reorg case here, so user don't need to handle it manually.
+        let (tip_height, _) = self.get_tip()?;
+        let reorged_headers = self
+            .block_headers((tip_height - 20) as usize, 21)?
+            .headers
+            .into_iter()
+            .map(|header| header.block_hash())
+            .zip((tip_height - 20)..=tip_height);
+
+        // Insert the new checkpoints
+        for (hash, height) in reorged_headers {
+            if sparse_chain
+                .insert_checkpoint(BlockId { height, hash })
+                .is_err()
+            {
+                return Err(ElectrumError::Reorg(height));
+            };
+        }
```

Comment:
Simplify this. I think we should just check that our tip hasn't changed here. If it has return the re-org error. That's enough for now.